packages/server/graphql/public/mutations/updateCreditCard.ts

```diff
@@ -0,0 +1,104 @@
+import standardError from '../../../utils/standardError'
+import {getStripeManager} from '../../../utils/stripe'
+import {stripeCardToDBCard} from '../../mutations/helpers/stripeCardToDBCard'
+import {MutationResolvers} from '../resolverTypes'
+
+const updateCreditCard: MutationResolvers['updateCreditCard'] = async (
+  _source,
+  {orgId, paymentMethodId},
+  {authToken, dataLoader, socketId: mutatorId}
+) => {
+  const operationId = dataLoader.share()
+  const subOptions = {mutatorId, operationId}
+  const now = new Date()
+  const r = await getRethink()
+
+  // AUTH
+  const viewerId = getUserId(authToken)
+  if (!(await isUserBillingLeader(viewerId, orgId, dataLoader))) {
+    return standardError(new Error('Must be the organization leader'), {userId: viewerId})
+  }
+
+  // RESOLUTION
+  const organization = await dataLoader.get('organizations').load(orgId)
+  const {stripeId, stripeSubscriptionId} = organization
+  if (!stripeId || !stripeSubscriptionId) {
+    return standardError(new Error('Organization is not subscribed to a plan'), {userId: viewerId})
+  }
+  const manager = getStripeManager()
+  const subscription = await manager.retrieveSubscription(stripeSubscriptionId)
+  if (subscription instanceof Error) return standardError(subscription, {userId: viewerId})
+  if (!subscription) {
+    return standardError(new Error('Organization is not subscribed to a plan'), {
+      userId: viewerId
+    })
+  }
+  const customer = await manager.retrieveCustomer(stripeId)
+  if (customer.deleted) {
+    return standardError(new Error('Stripe customer has been deleted'), {userId: viewerId})
+  }
+  const {id: customerId} = customer
+  const res = await manager.attachPaymentToCustomer(customerId, paymentMethodId)
+  if (res instanceof Error) return standardError(res, {userId: viewerId})
+  await manager.updateDefaultPaymentMethod(customerId, paymentMethodId)
+  await manager.updateSubscription(subscription.id, paymentMethodId)
+  const stripeCard = await manager.retrieveCardDetails(paymentMethodId)
+  if (stripeCard instanceof Error) return standardError(stripeCard, {userId: viewerId})
+  const creditCard = stripeCardToDBCard(stripeCard)
+
+  try {
+    await Promise.all([
+      r({
+        updatedOrg: r.table('Organization').get(orgId).update({
+          creditCard,
+          tier: 'team',
+          stripeId: customer.id,
+          tierLimitExceededAt: null,
+          scheduledLockAt: null,
+          lockedAt: null,
+          updatedAt: now
+        })
+      }).run(),
+      updateTeamByOrgId(
+        {
+          isPaid: true,
+          tier: 'team'
+        },
+        orgId
+      ),
+      removeTeamsLimitObjects(orgId, dataLoader)
+    ])
+    await Promise.all([setUserTierForOrgId(orgId), setTierForOrgUsers(orgId)])
+  } catch (e) {
+    const error =
+      e instanceof Error ? e : new Error('Failed to update db after updating credit card')
+    return standardError(error, {userId: viewerId, tags: {orgId}})
```

Comment:
We're calling five separate functions here, so I thought I'd add a try catch just in case something goes wrong. Happy to remove it if it's just confusing

Reply:
Let's remove it. We're not expecting any failure here, and if we experience one, we'll get an exception already